x-element.js

```diff
@@ -1662,47 +1688,3 @@ class Template {
     }
   }
 }
-
-/** Internal implementation details for template results. */
-class TemplateResult {
```

Comment:
Pretty much all of the diff here is a flattening / renaming of concepts. I found myself squinting to explain why we have a function called `template`, a class called the `TemplateEngine`, a class called `Template` and a class called `TemplateResult`…

_Now_, we can (roughly) say the following:

> The “template” function returns a “template result” by leveraging a “template engine”.

^^ As the saying goes the wrong abstraction is worse than no abstraction at all! Rather than have all those extra classes, we’re en route to just having `TemplateEngine`.